broadpeakplugin/src/main/java/com/kaltura/playkit/plugins/broadpeak/BroadpeakPlugin.java

```diff
@@ -112,6 +139,14 @@ public void apply(PKMediaEntry mediaEntry, PKMediaEntryInterceptor.Listener list
 
         if (mediaEntry != null && mediaEntry.getSources() != null &&
                 !mediaEntry.getSources().isEmpty() && mediaEntry.getSources().get(0) != null) {
+
+            // Stop the session for fresh media entry
+            if (lastMediaEntry != null) {
```

Comment:
@GouravSna  worth checking change media to same media what is the behaviour if we have to verify if ew equals to last